src/java.base/share/classes/java/lang/invoke/X-VarHandleSegmentView.java.template

```diff
@@ -97,9 +97,9 @@ final class VarHandleSegmentAs$Type$s extends VarHandleSegmentViewBase {
 #end[floatingPoint]
 
     @ForceInline
-    static AbstractMemorySegmentImpl checkAddress(Object obb, long offset, long length, boolean ro) {
+    static AbstractMemorySegmentImpl checkReadOnly(Object obb, boolean ro) {
```

Comment:
This and the following methods are the bulk of the changes in this template. That is, we no longer check size and alignment of the accessed segment. Every other change in this template is needed to get rid of fields and parameters that are no longer used.

Reply:
Note: we don't even check for overflows, e.g. `offset < 0`. The reasoning here is that all layouts check for overflow on construction, so it's never possible to construct a layout whose size is greater than `Long.MAX_VALUE`. And, since all the `long` indices corresponding to open path elements are checked against their bounds (and their bounds must be small enough so that the enclosing layout has a size smaller than `Long.MAX_VALUE`), and since we already checked (outside) that the segment is big enough to hold the enclosing layout, then the offset we see here must always point somewhere inside the segment.